osf/models/external.py

```diff
@@ -150,12 +151,23 @@ def auth_url(self):
             session.data['oauth_states'] = {}
 
         if self._oauth_version == OAUTH2:
+            # Quirk: Some time between 2019/05/31 and 2019/06/04, Bitbucket's OAuth2 API no longer
+            #        expects the query param `redirect_uri` in the `oauth2/authorize` endpoint.  In
+            #        addition, it relies on the "Callback URL" of the "OAuth Consumer" to redirect
+            #        the auth flow after successful authorization.  `ADDONS_OAUTH_NO_REDIRECT` is a
+            #        list containing addons that do not use `redirect_uri` in OAuth2 requests.
+            if self.short_name in ADDONS_OAUTH_NO_REDIRECT:
+                redirect_uri = None
+            else:
+                redirect_uri = web_url_for(
```

Comment:
Not sure which one is the preferred style: the `if ... else ...` I have, or either the following two:

```python
redirect_uri = None
if self.short_name not in ADDONS_OAUTH_NO_REDIRECT:
    redirect_uri = web_url_for()
```

```python
redirect_uri = web_url_for()
if self.short_name in ADDONS_OAUTH_NO_REDIRECT:
    redirect_uri = None
```

There is another occurrence in `ExternalProvider.auth_callback`.

Reply:
I don't have strong preferences, other than preferring "in" instead of "not in", as I think it's more readable.   And in going with the "in" formatting, I like how this doesn't build web_url_for() unless necessary, so I think your choice is fine.